signing/mcl/multisig/bls_bench_test.go

```diff
@@ -0,0 +1,184 @@
+package multisig_test
+
+import (
+	"testing"
+
+	crypto "github.com/ElrondNetwork/elrond-go-crypto"
+	"github.com/ElrondNetwork/elrond-go-crypto/mock"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing/mcl"
+	"github.com/ElrondNetwork/elrond-go-crypto/signing/mcl/multisig"
+	"github.com/herumi/bls-go-binary/bls"
+	"github.com/stretchr/testify/require"
+)
+
+func Benchmark_PreparePublicKeys63(b *testing.B) {
+	benchmarkPreparePublicKeys(63, b)
+}
+
+func Benchmark_PreparePublicKeys400(b *testing.B) {
+	benchmarkPreparePublicKeys(400, b)
+}
+
+func benchmarkPreparePublicKeys(nPubKeys int, b *testing.B) {
+	hasher := &mock.HasherSpongeMock{}
+
+	pubKeys := createBLSPubKeys(nPubKeys)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		prepPubKeys, err := multisig.PreparePublicKeys(pubKeys, hasher, pubKeys[0].Suite())
+		require.Nil(b, err)
+		require.NotNil(b, prepPubKeys)
+	}
+}
+
+func Benchmark_ConcatPubKeys63(b *testing.B) {
+	benchmarkConcatPubKeys(63, b)
+}
+
+func Benchmark_ConcatPubKeys400(b *testing.B) {
+	benchmarkConcatPubKeys(400, b)
+}
+
+func benchmarkConcatPubKeys(nPubKeys int, b *testing.B) {
+	pubKeys := createBLSPubKeys(nPubKeys)
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		_, err := multisig.ConcatPubKeys(pubKeys)
+		require.Nil(b, err)
+	}
+}
+
+func Benchmark_VerifyAggregatedSig63(b *testing.B) {
+	benchmarkVerifyAggregatedSig(63, b)
+}
+
+func Benchmark_VerifyAggregatedSig400(b *testing.B) {
+	benchmarkVerifyAggregatedSig(400, b)
+}
+
+func benchmarkVerifyAggregatedSig(nPubKeys uint16, b *testing.B) {
+	msg := []byte(testMessage)
+
+	hasher := &mock.HasherSpongeMock{}
```

Comment:
can we use the real components for benchmarks?
check the other occurrences

Reply:
changed to use blake2b hasher